modules/misc/nixgl.nix

```diff
@@ -4,30 +4,75 @@ let cfg = config.nixGL;
 in {
   meta.maintainers = [ lib.maintainers.smona ];
 
-  options.nixGL.prefix = lib.mkOption {
-    type = lib.types.str;
-    default = "";
-    example = lib.literalExpression
-      ''"''${inputs.nixGL.packages.x86_64-linux.nixGLIntel}/bin/nixGLIntel"'';
-    description = ''
-      The nixGL command that `lib.nixGL.wrap` should wrap packages with.
-      This can be used to provide libGL access to applications on non-NixOS systems.
-
-      Some packages are wrapped by default (e.g. kitty, firefox), but you can wrap other packages
-      as well, with `(config.lib.nixGL.wrap <package>)`. If this option is empty (the default),
-      then `lib.nixGL.wrap` is a no-op.
-    '';
+  options.nixGL = {
+    packages = lib.mkOption {
+      type = with lib.types; nullOr attrs;
+      default = null;
+      example = lib.literalExpression "inputs.nixGL.packages";
+      description = ''
+        The nixGL package set containing GPU library wrappers. This can be used to
+        provide OpenGL and Vulkan access to applications on non-NixOS systems by
+        using `(config.lib.nixGL.wrapMesa <package>)` or
+        `(config.lib.nixGL.wrapNvidia <package>)` for free (Intel, AMD, nouveau,
+        ...) and proprietary (Nvidia) drivers, respectively.
+
+        The `wrapMesa` and `wrapNvidia` functions are always available. If this
+        option is empty (the default), they are a no-op.
+
+        Note that using the Nvidia wrapper requires building the configuration
+        with the `--impure` option.
+      '';
+    };
+
+    nvidiaOffload = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        When enabled, the Nvidia nixGL wrapper will set up the environment for
+        PRIME offloading.
+
+        Additionaly, the `nvidia-offload` script will be installed to allow
+        starting programs on the dGPU manually. This script is useful for
+        programs not installed via Home Manager.
+      '';
+    };
+
+    wrapperScripts = lib.mkOption {
+      type = lib.types.enum [ "none" "mesa" "nvidia" "both" ];
+      default = "none";
+      description = ''
+        When set to a value other than "none", nixGL wrappers are made
+        available as shell commands, as follows:
+
+        - "mesa" enables the `nixGLMesa` script;
+        - "nvidia" enables the `nixGLNvidia` script, which honors the
+          `nixGL.nvidiaOffload` option;
+        - "both" enables both of the above.
+
+        These scripts are useful for programs not installed via Home Manager.
+      '';
+    };
   };
 
-  config = {
-    lib.nixGL.wrap = # Wrap a single package with the configured nixGL wrapper
-      pkg:
+  config = let
+    getWrapperExe = vendor: let
+      glPackage = cfg.packages.${pkgs.system}."nixGL${vendor}";
+      vulkanPackage = cfg.packages.${pkgs.system}."nixVulkan${vendor}";
+    in "${lib.getExe glPackage} ${lib.getExe vulkanPackage}";
```

Comment:
why are you wrapping the process in both the glPackage and the vulkanPackage? shouldn't that be up to the user whether they want to use closed or open source drivers?

Reply:
It's not about closed or open drivers. Vulkan and OpenGL are both graphics APIs, and it's up to the program which they use. I don't think the user should need to know which API the program uses, especially when the program can use either, as is the case of mpv or Chromium. See also https://github.com/nix-community/nixGL/issues/91

Reply:
Honestly, I don't know why nixGL offers separate wrappers for OpenGL and Vulkan.